### PR TITLE
Add Flow

### PR DESCRIPTION
**Pull request by nhunzaker**
This commit adds FlowType to Microcosm. Additionally, it implements
types for the internal archive package.

Away we go.

### CODE REVIEW

File: src/archive.js [hunk 1]
```diff
@@ -1,9 +1,16 @@
 /**
  * @fileoverview Keeps track of prior action states according to an
  * action's id
+ * @flow
  */
 
+import type Action from './action'
+
+type Payload = { [string]: * }
```
Comment (leobauza) [inline src/archive.js]: Can you explain this bit to me? So you are defining a `Payload` type and it is an object that has...and then I don't know what the rest means.

File: src/archive.js [hunk 2]
```diff
@@ -16,36 +23,45 @@ class Archive {
...
+    return typeof this.pool[action.id] !== 'undefined'
   }
 
   /**
    * Assign a new snapshot for an action
    * @param {Action} action Action for requested snapshot
    * @param {Object} snapshot
    */
-  set(action, snapshot) {
+  set(action: Action, snapshot: Payload) {
```
Comment (leobauza) [inline src/archive.js]: You have `remove()` returning `void` but this unspecified. Does flow automatically assume `void` as a return value or should we add that here or take it off of `remove()`

Comment (nhunzaker) [inline src/archive.js]: Ah, good catch. I believe it does. I think we should just omit `:void`.

File: src/archive.js (hunk 1 above)
Comment (nhunzaker) [inline src/archive.js]: Yeah. It is an object where the keys are strings and the values could be anything.

Comment (nhunzaker) [inline src/archive.js]: Actually... That's pretty useless. I'll just reduce this down to Object.

**Pull request merged**